lib/daru/io/io.rb

```diff
@@ -201,25 +199,29 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
           ::CSV
-          .open(path, 'rb', opts)
+          .parse(new_open(path), opts)
           .tap { |c| yield c if block_given? }
           .to_a
-
         headers       = ArrayHelper.recode_repeated(csv_as_arrays.shift)
         csv_as_arrays = csv_as_arrays.transpose
-
         headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
       end
+
+      def new_open(url)
+        open(url)
+      rescue URI::InvalidURIError
+        host = url.match(".+\:\/\/([^\/]+)")[1]
+        Net::HTTP.get host, url.partition(host)[2] || '/'
```

Comment:
I do understand that there won't be use for this trick / workaround from Ruby 2.2 onwards. And yes, we could address any issue about domains with underscore, but then the user will have to have the workaround in their code would have to override our existing #from_csv function - which is what I was trying to avoid. I was trying to avoid any hassle that the user may face, but this works too. :smile: